docs/runtime-contract.md

```diff
@@ -1,19 +1,75 @@
 similar but way simpler to this: https://github.com/knative/serving/blob/master/docs/runtime-contract.md
 
-We can define the general shared runtime contract for Source the interface:
+# Knative Runtime Contract
+
+## Background
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" are to be
+interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+This document considers two users of a given Knative environment, and is
+particularly concerned with the expectations of developers (and language and
+tooling developers, by extension) running code in the environment.
+
+ - **Developers** write code which is packaged into a container which is run on the
+   Knative cluster.
+   - **Language and tooling developers** typically write tools used by developers to
+     package code into containers. As such, they are concerned that tooling
+     which wraps developer code complies with this runtime contract.
+ - **Operators** (also known as platform providers) provision the compute resources
+   and manage the software configuration of Knative and the underlying
+   abstractions (for example, Linux, Kubernetes, Istio, etc).
 
-- `K_SINK` <-- URI to send to.
-- `K_OUTPUT_FORMAT` <-- "binary" or "structured"
-TODO: extra Sources stuff.
-  
-- Source 
 
+## Environment
+
+Containers must be started with the following environment variables set:
+
+| Name              | Value                                                |
+| ---               | ---                                                  |
+| `K_SINK`          | This will be a URI.                                  |
+| `K_OUTPUT_FORMAT` | This will be one of either `structured` or `binary`. |
+
+TODO: extra Sources stuff.
 
-DataPlane contract :
+## Runtime & Lifecycle
 
-- HTTP POST of CloudEvents to `K_SINK`. (any version, please in `K_OUTPUT_FORMAT` if possible)
+Containers written by developers are subject to the following:
+ - A container may be killed if the URI of the sink changes.
+ - The container must send CloudEvents to the URI specified in `K_SINK`.
+ - The container should send CloudEvents with structured or binary encoding
+   matching `K_OUTPUT_FORMAT`.
+ - The container should send CloudEvents over HTTP POST.
+   - Note that the sink does not necessarily have to have the scheme `http` or
+     `https`, but this is not the standard use case.
```

Comment:
`but this is the standard use case.`

Reply:
clarified "http is the standard use case", no ambiguous "this"